backend/src/AppServer.hs

```diff
@@ -255,13 +256,16 @@ submitReportHandler (SubmitReportRequest rawReport logFileData) = do
       pure SubmitGameReportResponse {report = processedReport, winnerRating, loserRating}
 
 getReportsHandler :: Maybe Int64 -> Maybe Int64 -> AppM GetReportsResponse
-getReportsHandler limit offset = GetReportsResponse . map fromGameReport <$> runDb (getGameReports limit' offset')
-  where
-    maxLimit = 500
-    (limit', offset') = case (limit, offset) of
-      (Nothing, _) -> (maxLimit, 0)
-      (Just lim, Nothing) -> (lim, 0)
-      (Just lim, Just off) -> (lim, off)
+getReportsHandler limit offset =
+    runDb getNumGameReports >>= \total ->
```

Comment:
Very nice! Please do feel empowered to use do-syntax though, there are good reasons it exists! And one of those reasons is the awkwardness of requiring a bunch of nested binds when dealing with the results of multiple sequentially applied actions. 

What you had before wasn't wrong, and this isn't wrong either. Now that you see how this works, you are empowered to choose 😊

Reply:
All the power 💪 💪